packages/manager/src/features/CloudPulse/Alerts/CreateAlert/CreateAlertDefinition.tsx

```diff
@@ -0,0 +1,294 @@
+import { APIError } from '@linode/api-v4';
+import { CreateAlertDefinitionPayload } from '@linode/api-v4/lib/cloudpulse/types';
+import { createAlertDefinitionSchema } from '@linode/validation';
+import { ErrorMessage, FormikProvider, useFormik } from 'formik';
+import { useSnackbar } from 'notistack';
+import * as React from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { ActionsPanel } from 'src/components/ActionsPanel/ActionsPanel';
+import { Autocomplete } from 'src/components/Autocomplete/Autocomplete';
+import { Box } from 'src/components/Box';
+import { Breadcrumb } from 'src/components/Breadcrumb/Breadcrumb';
+import { Drawer } from 'src/components/Drawer';
+import { Notice } from 'src/components/Notice/Notice';
+import { Paper } from 'src/components/Paper';
+import { TextField } from 'src/components/TextField';
+import { Typography } from 'src/components/Typography';
+import { useCreateAlertDefinition } from 'src/queries/cloudpulse/alerts';
+import { getErrorMap } from 'src/utilities/errorUtils';
+import {
+  handleFieldErrors,
+  handleGeneralErrors,
+} from 'src/utilities/formikErrorUtils';
+
+import { AlertSeverityOptions } from '../constants';
+import { AddChannelListing } from './AddChannelListing';
+import { AddNotificationChannel } from './AddNotificationChannel';
+import { MetricCriteriaField } from './Custom/Metrics/MetricCriteria';
+import { TriggerConditions } from './Custom/TriggerConditions';
+import { EngineOption } from './shared/EngineOption';
+import { CloudPulseRegionSelect } from './shared/RegionSelect';
+import { CloudPulseMultiResourceSelect } from './shared/ResourceMultiSelect';
+import { CloudPulseServiceSelect } from './shared/ServicetypeSelect';
+import { useDatabaseEnginesQuery } from 'src/queries/databases';
+
+export const initialValues: CreateAlertDefinitionPayload = {
+  alertName: '',
+  criteria: [
+    {
+      aggregationType: '',
+      filters: [],
+      metric: '',
+      operator: '',
+      value: 0,
+    },
+  ],
+  engineOption: '',
+  notifications: [],
+  region: '',
+  resourceId: [],
+  serviceType: '',
+  severity: '',
+  triggerCondition: {
+    criteriaCondition: '',
+    evaluationPeriod: '',
+    pollingInterval: '',
+    triggerOccurrence: 0,
+  },
+};
+
+export const CreateAlertDefinition = React.memo(() => {
+  const [openAddNotification, setOpenAddNotification] = React.useState(false);
+  const [notifications, setNotifications] = React.useState<any>([]);
+  const { mutateAsync } = useCreateAlertDefinition();
+  const { enqueueSnackbar } = useSnackbar();
+  const { data: engineOptions } = useDatabaseEnginesQuery(true); 
+
+  const onSubmitAddNotification = (notification: any) => {
```

Comment:
'any' type is used in many places...i guess it makes code loosely typed..can we make it more specific..let me know if I am wrong here